.github/workflows/release-code-freeze.yml

```
@@ -21,142 +21,142 @@ env:
-        id: notify
-        with:
-          slack-bot-user-oauth-access-token: ${{ secrets.CODE_FREEZE_BOT_TOKEN }}
-          slack-channel: ${{ inputs.slackChannelOverride || secrets.WOO_RELEASE_SLACK_CHANNEL }}
-          slack-text: |
-            :warning-8c: ${{ needs.maybe-create-next-milestone-and-release-branch.outputs.release_version }} Code Freeze :ice_cube:
-            
-            The automation to cut the release branch for ${{ needs.maybe-create-next-milestone-and-release-branch.outputs.release_version }} has run. Any PRs that were not already merged will be a part of ${{ needs.maybe-create-next-milestone-and-release-branch.outputs.next_version }} by default. If you have something that needs to make ${{ needs.maybe-create-next-milestone-and-release-branch.outputs.release_version }} that hasn't yet been merged, please see the <${{ secrets.FG_LINK }}/code-freeze-for-woocommerce-core-release/|fieldguide page for the code freeze>.
-
-  trigger-changelog-action:
-    name: "Trigger changelog action"
-    runs-on: ubuntu-20.04
-    needs: maybe-create-next-milestone-and-release-branch
-    steps:
-      - run: |
-          curl \
-            -X POST \
-            -H "Accept: application/vnd.github.v3+json" \
-            -H "Authorization: token ${{ secrets.WC_BOT_TRIAGE_TOKEN }}" \
-            -d '{"ref":"refs/heads/trunk","inputs":{"releaseBranch":"${{ needs.maybe-create-next-milestone-and-release-branch.outputs.branch }}","releaseVersion":"${{ needs.maybe-create-next-milestone-and-release-branch.outputs.release_version }}"}}' \
-            https://api.github.com/repos/${{ github.repository }}/actions/workflows/release-changelog.yml/dispatches
+    verify-code-freeze:
+        name: 'Verify that today is the day of the code freeze'
+        runs-on: ubuntu-20.04
+        outputs:
+            freeze: ${{ steps.check-freeze.outputs.freeze }}
+        steps:
+            - name: 'Install PHP'
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: '7.4'
+
+            - name: 'Check whether today is the code freeze day'
+              id: check-freeze
+              shell: php {0}
+              run: |
+                  <?php
+                  $now = time();
+                  if ( getenv( 'TIME_OVERRIDE' ) ) {
+                    $now = strtotime( getenv( 'TIME_OVERRIDE' ) );
+                  }
+
+                  // Code freeze comes 26 days prior to release day.
+                  $release_time         = strtotime( '+26 days', $now );
+                  $release_day_of_week  = date( 'l', $release_time );
+                  $release_day_of_month = (int) date( 'j', $release_time );
+
+                  // If 26 days from now isn't the second Tuesday, then it's not code freeze day.
+                  if ( 'Tuesday' !== $release_day_of_week || $release_day_of_month < 8 || $release_day_of_month > 14 ) {
+                    echo 'freeze=1' >> $GITHUB_OUTPUT;
```

This isn't going to work, as it's PHP code being executed.

This might be worth a refactor, or a simple solution might be something like:

```suggestion
                    file_put_contents( getenv( 'GITHUB_OUTPUT') , "freeze=1\n", FILE_APPEND );
```

(I haven't tested this)

---

Oh shoot you're right. ugh. ok thanks, yeah this code example looks pretty good!